client/src/design-system/components/editor/plugins/images-plugin.tsx

```diff
@@ -0,0 +1,383 @@
+          Confirm
+        </Button>
+      </DialogFooter>
+    </div>
+  );
+};
+
+export const InsertImageUploadedDialogBody = ({
+  onClick,
+}: {
+  onClick: (payload: InsertImagePayload) => void;
+}) => {
+  const [src, setSrc] = useState("");
+  const [altText, setAltText] = useState("");
+
+  const isDisabled = src === "";
+
+  const loadImage = (files: FileList | null) => {
+    const reader = new FileReader();
+    reader.onload = () => {
+      if (typeof reader.result === "string") {
+        setSrc(reader.result);
+      }
+      return "";
+    };
+    if (files !== null) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      reader.readAsDataURL(files[0]);
+    }
+  };
+
+  return (
+    <div className="grid gap-4 py-4">
+      <div className="grid gap-2">
+        <Label htmlFor="image-upload">Image Upload</Label>
+        <Input
+          id="image-upload"
+          type="file"
+          onChange={(e) => loadImage(e.target.files)}
+          accept="image/*"
+          data-test-id="image-modal-file-upload"
+        />
+      </div>
+      <div className="grid gap-2">
+        <Label htmlFor="alt-text">Alt Text</Label>
+        <Input
+          id="alt-text"
+          placeholder="Descriptive alternative text"
+          onChange={(e) => setAltText(e.target.value)}
+          value={altText}
+          data-test-id="image-modal-alt-text-input"
+        />
+      </div>
+      <Button
+        type="submit"
+        disabled={isDisabled}
+        onClick={() => onClick({ altText, src })}
+        data-test-id="image-modal-file-upload-btn"
+      >
+        Confirm
+      </Button>
+    </div>
+  );
+};
+
+export const InsertImageDialog = ({
+  activeEditor,
+  onClose,
+}: {
+  activeEditor: LexicalEditor;
+  onClose: () => void;
+}): JSX.Element => {
+  const hasModifier = useRef(false);
+
+  useEffect(() => {
+    hasModifier.current = false;
+    const handler = (e: KeyboardEvent) => {
+      hasModifier.current = e.altKey;
+    };
+    document.addEventListener("keydown", handler);
+    return () => {
+      document.removeEventListener("keydown", handler);
+    };
+  }, [activeEditor]);
+
+  const onClick = (payload: InsertImagePayload) => {
+    activeEditor.dispatchCommand(INSERT_IMAGE_COMMAND, payload);
+    onClose();
+  };
+
+  return (
+    <Tabs defaultValue="url">
+      <TabsList className="w-full">
+        <TabsTrigger value="url" className="w-full">
+          URL
+        </TabsTrigger>
+        <TabsTrigger value="file" className="w-full">
+          File
+        </TabsTrigger>
+      </TabsList>
+      <TabsContent value="url">
+        <InsertImageUriDialogBody onClick={onClick} />
+      </TabsContent>
+      <TabsContent value="file">
+        <InsertImageUploadedDialogBody onClick={onClick} />
+      </TabsContent>
+    </Tabs>
+  );
+};
+
+export const ImagesPlugin = ({
+  captionsEnabled,
+}: {
+  captionsEnabled?: boolean;
+}): JSX.Element | null => {
```

Comment:
je comprends pas pq iels ont typé tous les retours de composants wtf

```suggestion
}) => {
```